cli/src/main/java/denominator/cli/Denominator.java

```diff
@@ -426,6 +428,32 @@ void overrideFromEnv(Map<String, String> env) {
       }
     }
 
+    static void setProxyFromEnv() {
+      setProtocolProxyFromEnv("http");
+      setProtocolProxyFromEnv("https");
+    }
+
+    static void setProtocolProxyFromEnv(String proto) {
+      String lowerProto = Ascii.toLowerCase(proto);
+      String envProxy = System.getenv(Ascii.toUpperCase(proto) + "_PROXY");
```

Comment:
if you move this to a parameter, then you don't use System.getenv inside this method. That makes it testable with system rules!